scripts/solve.js

```diff
@@ -6,14 +6,29 @@ window.addEventListener('load', function() {
     mode: 'javascript',
     lineNumbers: true
   });
-
+  
   runStaticAnalysis(myCodeMirror.getValue());
+  
+  myCodeMirror.on('change', function() {
+    runStaticAnalysis(myCodeMirror.getValue());
+  });
 });
 
 function runStaticAnalysis(code) {
   let totalComplexity = 0;
   JSHINT(code);
   const results = JSHINT.data();
   results.functions.forEach(fn => totalComplexity += fn.metrics.complexity);
-  document.getElementById('analysis-output').innerText = 'The cyclomatic complexity is ' + totalComplexity;
+
+  functionNames = results.functions.map(element => element.name);
+  unusedVarNames = results.unused.map(element => element.name);
+  
+  // Only show unused variables, not functions
+  let unusedVars = unusedVarNames.filter(x => !functionNames.includes(x));
```

Comment:
it seems like you do this to exclude the main function.

what if there are unused functions? Is there a way to exclude only the main function?

Reply:
i wasn't sure how best to go about it this - i was thinking that we could specify to the user that the main function should be called `solution` and only exclude that, does that sound reasonable?

Reply:
I personally think that sounds reasonable.
Also, this might not even be a problem as we could include sample test cases in problems, which will use the main function.

Reply:
perhaps leaving a TODO and deciding after we have clarity on how test cases will be provided and how they are handled by JSHINT would be ideal.